ipaserver/plugins/cert.py

```diff
@@ -1101,15 +1101,22 @@ def get_params(self):
             )
 
     def _owners(self):
-        for name in ('user', 'host', 'service'):
-            yield self.api.Object[name]
+        for obj_name, pkey_name in [('user', None),
+                                    ('host', None),
+                                    ('service', 'krbprincipalname')]:
+            obj = self.api.Object[obj_name]
+            if pkey_name is None:
+                pkey = obj.primary_key
+            else:
+                pkey = obj.params[pkey_name]
+            yield obj, pkey
 
     def _fill_owners(self, obj):
         dns = obj.pop('owner', None)
         if dns is None:
             return
 
-        for owner in self._owners():
+        for owner, __ in self._owners():
```

Comment:
I'm used to use `_` to variables that are not used, but since `_` is a function, I used `__`. Not sure if this is the right approach.